File: Karpa/Karpa_lab4/GAEvolve.java
package com.lab4;

public class GAEvolve {
	public float[] best_fitness;
	public GAIndividual best_ind;

	public GAEvolve(int generations, int pop_size, int genome_size, int xrate,
			int mrate, float[] min_range, float[] max_range) {
		best_fitness = new float[generations];
		GAPopulation gap = new GAPopulation(pop_size, genome_size, min_range,
				max_range);
		best_fitness[0] = gap.ind[gap.best_index].fitness;

		for (int i = 1; i < generations; i++) {
			gap = gap.generate(gap, xrate, mrate, min_range, max_range);
			best_fitness[i] = gap.ind[gap.best_index].fitness;
			System.out.println("Найкраще значення покоління " + best_fitness[i]);
		}
		best_ind = gap.ind[gap.best_index];
	}

	public static void main(String[] args) {
		// Min
		// float[] min = new float[] { -5f };
		// float[] max = new float[] { 4f };
		// Max
		float[] min = new float[]{15f};
		float[] max = new float[]{25f};
		GAEvolve gae = new GAEvolve(10, 10, 1, 70, 20, min, max);
		System.out.println("Результат:");
		System.out.println(gae.best_ind);

	}
}

File: Turash/Turash_lab4/source/turash_lab4(csharp)/turash_lab4/GAEvolve.cs
using System;

namespace turash_lab4
{
    class GAEvolve
    {
        public double[] BestFitness;//є найкращим значенням для і-го покоління даної еволюції
        public GAIndividual BestInd; //найкращий індивідум(х) даної еволюції
        public GAEvolve(int generations, int popSize, int genomeSize, int xrate, int mrate, double[] minRange, double[] maxRange)
        {

            BestFitness = new double[generations];
            var gap = new GAPopulation(popSize, genomeSize, minRange, maxRange);
            BestFitness[0] = gap.Ind[gap.BestIndex].Fitness;

            for (var i = 1; i < generations; i++)
            {
                gap = gap.Generate(gap, xrate, mrate, minRange, maxRange);
                BestFitness[i] = gap.Ind[gap.BestIndex].Fitness;

                Console.WriteLine("Найкраще значення поколiння " + BestFitness[i]);
            }
            BestInd = gap.Ind[gap.BestIndex];
        }

        static void Main(string[] args)
        {

            double[] min = { -10f };
            double[] max = { 53 };
            var gae = new GAEvolve(100, 5, 1, 70, 20, min, max);
            Console.WriteLine("Результат:");
            Console.WriteLine(gae.BestInd);
            Console.ReadKey();

        }
    }
}